### PR TITLE
Add account linking configs

### PR DESCRIPTION
**Pull request by tung2744**
ref https://github.com/authgear/authgear-server/issues/4111
ref DEV-1206

### CODE REVIEW

File: pkg/lib/config/authentication_flow.go [hunk 1]
```diff
@@ -624,6 +625,55 @@ var _ = Schema.Add("AuthenticationFlowAccountRecoveryIdentification", `
 }
 `)
 
+var _ = Schema.Add("AuthenticationFlowAccountLinking", `
+{
+	"type": "object",
+	"required": [],
```
Comment (louischan-oursky) [inline pkg/lib/config/authentication_flow.go:632]: Omit `required`.

File: pkg/lib/config/authentication_flow.go [hunk 2]
```diff
@@ -624,6 +625,55 @@ var _ = Schema.Add("AuthenticationFlowAccountRecoveryIdentification", `
...
+	"properties": {
+		"oauth": {
+			"type": "array",
+			"items": { "$ref": "#/$defs/AccountLinkingOAuth" }
```
Comment (louischan-oursky) [inline pkg/lib/config/authentication_flow.go:636]: Name the schema `AuthenticationFlowAccountLinkOAuthItem`

File: pkg/lib/config/authentication_flow.go [hunk 3]
```diff
@@ -624,6 +625,55 @@ var _ = Schema.Add("AuthenticationFlowAccountRecoveryIdentification", `
...
+}
+`)
+
+var _ = Schema.Add("AccountLinkingOAuth", `
+{
+	"type": "object",
+	"required": ["alias", "oauth_claim", "user_profile", "action"],
+	"properties": {
+		"alias": { "type": "string" },
+		"oauth_claim": { "$ref": "#/$defs/AccountLinkingJSONPointer" },
```
Comment (louischan-oursky) [inline pkg/lib/config/authentication_flow.go:648]: Name the schema `AuthenticationFlowAccountLinkingJSONPointer`

File: pkg/lib/config/authentication_flow.go [hunk 4]
```diff
@@ -624,6 +625,55 @@ var _ = Schema.Add("AuthenticationFlowAccountRecoveryIdentification", `
...
+		"user_profile": { "$ref": "#/$defs/AccountLinkingJSONPointer" },
+		"action": { "$ref": "#/$defs/AccountLinkingOAuthAction" }
```
Comment (louischan-oursky) [inline pkg/lib/config/authentication_flow.go]: Name the schema `AuthenticationFlowAccountLinkingAction`

File: pkg/lib/config/authentication_flow.go [hunk 5]
```diff
@@ -624,6 +625,55 @@ var _ = Schema.Add("AuthenticationFlowAccountRecoveryIdentification", `
...
+var _ = Schema.Add("AccountLinkingJSONPointer", `
+{
+	"type": "object",
+	"additionalProperties": false,
+	"properties": {
+		"pointer": {
+			"type": "string",
+			"format": "json-pointer"
+		}
+	}
```
Comment (louischan-oursky) [inline pkg/lib/config/authentication_flow.go:675]: Add `"required": ["pointer"]`

File: pkg/lib/config/authentication_flow.go [hunk 6]
```diff
@@ -1382,6 +1404,31 @@ const (
 	AuthenticationFlowAccountRecoveryIdentificationOnFailureIgnore = AuthenticationFlowAccountRecoveryIdentificationOnFailure("ignore")
 )
 
+type AuthenticationFlowAccountLinking struct {
+	OAuth []*AccountLinkingOAuth `json:"oauth,omitempty"`
+}
+
+type AccountLinkingOAuth struct {
+	Alias       string                    `json:"alias,omitempty"`
+	OAuthClaim  AccountLinkingJSONPointer `json:"oauth_claim,omitempty"`
```
Comment (louischan-oursky) [inline pkg/lib/config/authentication_flow.go:1395]: Should be a pointer to `AccountLinkingJSONPointer`

File: pkg/lib/config/authentication_flow.go [hunk 7]
```diff
@@ -1382,6 +1404,31 @@ const (
...
+	UserProfile AccountLinkingJSONPointer `json:"user_profile,omitempty"`
```
Comment (louischan-oursky) [inline pkg/lib/config/authentication_flow.go:1396]: Should be a pointer to `AccountLinkingJSONPointer`

**Review by louischan-oursky — COMMENTED** [review]
Let's add some tests on the newly added structs, like we did in `pkg/lib/config/testdata/signup_flow_tests.yaml`.

**Comment by tung2744** [comment]
Let me address the comments in https://github.com/authgear/authgear-server/pull/4159

**Pull request closed without merging**